taskmgr/task_manager.go

```diff
@@ -0,0 +1,60 @@
+package taskmgr
+
+import (
+	"github.com/Sirupsen/logrus"
+	tcqueue "github.com/taskcluster/taskcluster-client-go/queue"
+	"github.com/taskcluster/taskcluster-client-go/tcclient"
+	"github.com/taskcluster/taskcluster-worker/config"
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+)
+
+// Manager is resonsible for managing the entire task lifecyle from claiming the
+// task, creating a sandbox environment, and reporting the results fo the execution.
+// The manager will also be responsible for ensuring tasks do not run past their max run
+// time and are aborted if a cancellation message is received.
+type Manager struct {
+	// List of Tasks Contexts for running tasks
+	Tasks []*runtime.TaskContextController
+	// Maxmimum capacity that the worker is configured for.
+	MaxCapacity   int
+	Engine        *engines.Engine
+	Log           *logrus.Entry
+	Queue         *queueService
+	ProvisionerId string
+	WorkerGroup   string
+	WorkerId      string
+}
```

Comment:
`workerType` and `PluginManager` (or at least a todo to add the plugin manager).